transmart-batch/src/main/groovy/org/transmartproject/batch/concept/ConceptTree.groovy

```diff
@@ -115,6 +115,12 @@ class ConceptTree {
         if (node) {
             return node
         }
+        if(ClinicalVariable.conceptTypeFor(variable) != ConceptType.CATEGORICAL) {
+            /* to generate missing parent tree nodes for non-categorical variables */
+            for (def p = variable.path.parent; p != null; p = p.parent) {
```

Comment:
`getOrGenerate` already contains such a loop, so I think it is okay to just call the function on `variable.path.parent` (if not null).